src/sentry/utils/snuba.py

```diff
@@ -630,38 +635,50 @@ def raw_query(
         is_grouprelease=is_grouprelease,
         **kwargs,
     )
-    return bulk_raw_query([snuba_params], referrer=referrer)[0]
+    return bulk_raw_query([snuba_params], referrer=referrer, use_cache=use_cache)[0]
 
 
-def bulk_raw_query(snuba_param_list, referrer=None):
+def bulk_raw_query(snuba_param_list, referrer=None, use_cache=False):
     headers = {}
     if referrer:
         headers["referer"] = referrer
 
-    query_param_list = map(_prepare_query_params, snuba_param_list)
+    # Store the original position of the query so that we can maintain the order
+    query_param_list = list(enumerate(map(_prepare_query_params, snuba_param_list)))
 
-    def snuba_query(params):
-        query_params, forward, reverse, thread_hub = params
-        try:
-            with timer("snuba_query"):
-                referrer = headers.get("referer", "<unknown>")
-                if SNUBA_INFO:
-                    # We want debug in the body, but not in the logger, so dump the json twice
-                    logger.info(f"{referrer}.body: {json.dumps(query_params)}")
-                    query_params["debug"] = True
-                body = json.dumps(query_params)
-                with thread_hub.start_span(op="snuba", description=f"query {referrer}") as span:
-                    span.set_tag("referrer", referrer)
-                    for param_key, param_data in query_params.items():
-                        span.set_data(param_key, param_data)
-                    return (
-                        _snuba_pool.urlopen("POST", "/query", body=body, headers=headers),
-                        forward,
-                        reverse,
-                    )
-        except urllib3.exceptions.HTTPError as err:
-            raise SnubaError(err)
+    results = []
+
+    if use_cache:
+        # sqc - Snuba Query Cache
```

Comment:
Would this be something that could live in the snuba-sdk longer term?

Reply:
That's an interesting idea. I have been toying with the idea of moving some of the Snuba connection logic into the SDK, so maybe this would be one of those things.